src/components/generic/ApexLineChart.vue

```diff
@@ -0,0 +1,218 @@
+<template>
+  <div style="color: black;">
+    <vue-apex-charts
+      type="line"
+      height="350"
+      :options="chartOptions"
+      :series="_series"
+    ></vue-apex-charts>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import Component from 'vue-class-component';
+import VueApexCharts from 'vue-apexcharts';
+import {ApexOptions, exec} from 'apexcharts';
+import {install} from 'vuetify/es5/install';
+import {getModule} from 'vuex-module-decorators';
+import ColorHackModule from '@/store/color_hack';
+
+// We declare the props separately to make props types inferable.
+const ApexLineChartProps = Vue.extend({
+  props: {
+    categories: Array, // Should be of type string[]
+    series: Array, // Should be of type object[]
+    upper_range: Number, //upper bound of y axis
+    sev_chart: Boolean, //identifies chart as severity chart
+    title: String,
+    y_title: String
+  }
+});
+
+let id_counter = 0;
+function next_id(): number {
+  id_counter += 1;
+  return id_counter;
+}
+
+export interface SeriesItem {
+  name: string;
+  data: number[];
+}
+
+/**
+ * Emits "category-selected" with payload of type Category whenever a category is selected.
+ */
+@Component({
+  components: {
+    VueApexCharts
+  }
+})
+export default class ApexLineChart extends ApexLineChartProps {
+  chart_id: string = `line_chart_${next_id}`;
+
+  get _categories(): string[] {
+    // Ensure it's an array
+    if (!(this.categories instanceof Array)) {
+      throw new Error('series must be an array of strings');
+    }
+
+    // Ensure all of its elements are numbers
+    this.categories.forEach(element => {
+      if (typeof element !== 'string') {
+        throw new Error(`Invalid series item ${element}`);
+      }
+    });
+
+    // We now know the type is definitely number[]
+    let final = this.categories as string[];
+
+    // If we have any non-zero data, just returngive 0.01 of all
+    return final;
+  }
+
+  /**
+   * Provide a type-checked accessor to our series property
+   */
+  get _series(): SeriesItem[] {
+    // Ensure it's an array
+    if (!(this.series instanceof Array)) {
+      throw new Error('series must be an array of objects');
+    }
+
+    // Ensure all of its elements are numbers
+    for (let element of this.series) {
+      if (typeof element !== 'object') {
+        throw new Error(`Invalid series item ${element}`);
+      }
+    }
+
+    // We now know the type is definitely number[]
+    let final = this.series as SeriesItem[];
+
+    // If we have any non-zero data, just returngive 0.01 of all
+    return final;
+  }
+
+  get label_colors(): string[] {
+    let colors = [];
+    for (let i = 0; i < this._categories.length; i++) {
```

Comment:
vue-class-component does not give type checking, it assumes category is an array of any, so this is better solved when we implement the vue-property-decorator, many other components outside or inside my branch have this issue, so I think it might be better to fix it all at once in a different pr